docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+
+Note that we're skipping the ``allow_empty`` logic in ``ListView.get()``
+because it makes no sense to disallow empty listings for this view.
+
+We also need to decide what we should inherit from. We have the following
+options:
+
+* ``SingleObjectMixin``, ``MultipleObjectMixin`` and ``View``
+  (to provide view rendering methods).
+
+* ``DetailView`` and ``ListView``.
+
+* One of the views above and one of the Mixins for the other class.
+
+The two first options are both good, but the 3rd option is problematic because
+it makes it appear that we are using one of the view classes preferentially
+where in fact we just need the view methods available from either class. We'll
+use the first option in our example.
+
+We'll demonstrate this with the ``Author`` model we used in the
+:doc:`generic class-based views introduction<generic-display>`.
+
+.. snippet::
+    :filename: views.py
+
+    from django.views.generic import TemplateView
+    from django.views.generic.detail import SingleObjectMixin
+    from django.views.generic.list import MultipleObjectMixin
+    from books.models import Author
+
+    class AuthorDetail(SingleObjectMixin, MultipleObjectMixin, TemplateView):
+        """Combined detail / list view."""
+        model = Author
+        paginate_by = 2
+        template_name = "author.html"
+
+        def get(self, request, *args, **kwargs):
+            self.object = self.get_object()
+            self.object_list = self.object.books.all()
+            return self.render_to_response(self.get_context_data())
+
+We'll also have a ``Book`` model defined, linked to ``Author``:
+
+.. snippet::
+    :filename: models.py
+
+    from django import models
+
+    class Book(models.Model):
+        name = models.CharField(max_length=50, unique=True)
+        author = models.ForeignKey(Author, related_name="books")
+
+Note that when using ``context_object_name`` property or an automatically added
+context variable with the name based on model name, we need to be mindful of
+the inheritance order: in this case we'll use the ``author`` context variable
+which is set up in ``SingleObjectMixin.get_context_data()`` so we have to put
+``SingleObjectMixin`` first.
+
+.. code-block:: html+django
+
+    <h2>Author {{ author }}</h2>
+
+    <ol>
+      {% for book in page_obj %}
+        <li>{{ book.name }}</li>
+      {% endfor %}
+    </ol>
+
+    <div class="pagination">
+        <span class="step-links">
+            {% if page_obj.has_previous %}
+                <a href="?page={{ page_obj.previous_page_number }}">previous</a>
+            {% endif %}
+
+            <span class="current">
+                Page {{ page_obj.number }} of {{ paginator.num_pages }}.
+            </span>
+
+            {% if page_obj.has_next %}
+                <a href="?page={{ page_obj.next_page_number }}">next</a>
+            {% endif %}
+        </span>
+    </div>
+
+Combined Detail and Form Views
+==============================
```

Comment:
This example is overly simplified. Your `AuthorInterestForm` would need to save something most likely. It's not farfetched its validation would depend on the `Author` object in question in a more general use case and this would require more customisation. (Given the "Reuse of the self.object" point from the other PR this is a particularly pertinent example of why not to add that extra magic and make you override `get()` to do this).